x-pack/plugins/uptime/server/lib/synthetics_service/formatters/convert_to_data_stream.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DataStream, MonitorFields } from '../../../../common/runtime_types';
+
+interface DataStreamConfig {
+  type: DataStream;
+  id: string;
+  schedule: string;
+  data_stream: {
+    namespace: 'default';
+  };
+  streams: [
+    {
+      data_stream: {
+        dataset: DataStream;
+        type: 'synthetics';
+      };
+    } & Partial<MonitorFields>
+  ];
+}
+
+export function convertToDataStreamFormat(monitor: Record<string, any>): DataStreamConfig {
```

Comment:
Can we use the `SyntheticMonitor` type here, or nah?

Reply:
nah, since it's a formatted heartbeat config.

Reply:
Ah, makes sense. We could define that as `Record<ConfigKey, any>`, for a bit more typing